SharedContent/cpp-winui-template/WinUI3TemplateCpp/MainWindow.xaml.cpp

```diff
@@ -0,0 +1,101 @@
+﻿#include "pch.h"
+#include "MainWindow.xaml.h"
+#if __has_include("MainWindow.g.cpp")
+#include "MainWindow.g.cpp"
+
+#include "microsoft.ui.xaml.window.h" 
+#include "SampleConfiguration.h"
+#endif
+
+using namespace winrt;
+using namespace Microsoft::UI::Xaml;
+
+namespace winrt::WinUI3TemplateCpp::implementation
+{
+    MainWindow::MainWindow()
+    {
+        InitializeComponent();
+
+        Title(winrt::WinUI3TemplateCpp::Settings::FeatureName);
+
+        HWND hwnd = GetWindowHandle();
+
+        LoadIcon(hwnd, L"windows-sdk.ico");
+        SetWindowSize(hwnd, 1220, 1080);
+        PlacementCenterWindowInMonitorWin32(hwnd);
+    }
+
+    HWND MainWindow::GetWindowHandle()
+    {
+        if (_hwnd == nullptr)
+        {
+            Window window = this->try_as<Window>();
+            window.as<IWindowNative>()->get_WindowHandle(&_hwnd);
+        }
+        return _hwnd;
+    }
+
+    void MainWindow::LoadIcon(HWND hwnd, wchar_t* iconName)
+    {
+        auto hSmallIcon = LoadImage(NULL,
+            iconName,
+            IMAGE_ICON,
+            GetSystemMetrics(SM_CXSMICON),
+            GetSystemMetrics(SM_CYSMICON),
+            LR_LOADFROMFILE);
+        SendMessage(hwnd, WM_SETICON, ICON_SMALL, (LPARAM)hSmallIcon);
+
+        auto hBigIcon = LoadImage(
+            NULL,
+            iconName,
+            IMAGE_ICON,
+            GetSystemMetrics(SM_CXICON),
+            GetSystemMetrics(SM_CYICON),
+            LR_LOADFROMFILE);
+        SendMessage(hwnd, WM_SETICON, ICON_BIG, (LPARAM)hBigIcon);
+    }
+
+    void MainWindow::SetWindowSize(const HWND hwnd, const int width, const int height)
+    {
+        // Win32 uses pixels and WinUI 3 uses effective pixels, so you should apply the DPI scale factor
+        auto dpi = GetDpiForWindow(hwnd);
```

Comment:
Consider avoiding `auto` when the type isn't obvious. Particularly in light of the cast bug below, seeing that `dpi` is an integer rather than a `float` may help avoid such bugs.

Reply:
changed from `auto` to `UINT`